vault/provider.go

```diff
@@ -557,6 +581,30 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 
 	// Try an get the token from the config or token helper
 	token, err := providerToken(d)
+
+	// Attempt to use auth/<mount>login if 'auth_login' is provided in provider config
+	authLoginI := d.Get("auth_login").([]interface{})
+	if len(authLoginI) > 1 {
+		return "", fmt.Errorf("auth_login block may appear only once")
+	}
+
+	if len(authLoginI) == 1 {
+		authLogin := authLoginI[0].(map[string]interface{})
+		authLoginPath := authLogin["path"].(string)
+		authLoginNamespace := ""
+		if authLoginNamespaceI, ok := authLogin["namespace"]; ok {
+			authLoginNamespace = authLoginNamespaceI.(string)
+			client.SetNamespace(authLoginNamespace)
+		}
+		authLoginParameters := authLogin["parameters"].(map[string]interface{})
+
+		secret, err := client.Logical().Write(authLoginPath, authLoginParameters)
+		if err != nil {
+			return nil, err
+		}
+		token = secret.Auth.ClientToken
+	}
+
 	if err != nil {
```

Comment:
Should this err be checked immediately below `token, err := providerToken(d)` on line 583?

Reply:
You're right on that, i'll move this statement up in the file.